airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -662,30 +681,20 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
 
 
 class BulkIncrementalSalesforceStream(BulkSalesforceStream, IncrementalRestSalesforceStream):
-    def next_page_token(self, last_record: Mapping[str, Any]) -> Optional[Mapping[str, Any]]:
-        return None
+    state_checkpoint_interval = None
 
     def request_params(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
     ) -> MutableMapping[str, Any]:
-        start_date = max(
-            (stream_state or {}).get(self.cursor_field, ""),
-            (stream_slice or {}).get("start_date", ""),
-            (next_page_token or {}).get("start_date", ""),
-        )
+        start_date = max((stream_state or {}).get(self.cursor_field, ""), (stream_slice or {}).get("start_date", ""))
```

Comment:
this should handlet he case where both are `""`, the default value should be `2017-10-01` for this dev version

Reply:
seems like this is an old implementation, we always have a stream slices here and therefore `stream_slice.get("start_date")` will be enough